packages/react/spec/auto/inputs/PolarisAutoRelationshipInputBelongsTo.spec.tsx

```diff
@@ -0,0 +1,253 @@
+    const linkedSectionId = formSubmitRequest.variables.widget.section._link;
+    expect(linkedSectionId).toEqual("2");
+  });
+
+  test("it properly filters records based on the search value", async () => {
+    setupRenderResult({ searchValue: "Section Bar" }); // At this point there is no search so everything is shown
+    expect(screen.queryByText("Section Foo")).toBeInTheDocument();
+
+    await clickButton("setSearchButton"); // It is filtered out after applying the search
+    mockSectionQueryResponse([sectionBarEdge]); // Simulate search query response
+    expect(screen.queryByText("Section Foo")).toBeNull();
+
+    await clickButton("clearSearchButton"); // Clear the search to bring it back
+    mockSectionQueryResponse(); // Simulate search query response
+    expect(screen.queryByText("Section Foo")).toBeInTheDocument();
+    expect(screen.queryByText("Section Bar")).toBeInTheDocument();
+
+    await act(async () => {
+      // Select Foo
+      await user.click(screen.getByText("Section Foo"));
+    });
+
+    await clickButton("setSearchButton"); // Filter it out after it is already selected
+    mockSectionQueryResponse([sectionBarEdge]);
+    expect(screen.queryByText("Section Foo")).toBeNull();
+
+    await act(async () => {
+      await user.click(screen.queryByText("RealSubmitButton")!);
+    });
+
+    // Even though the search is applied and the selected value is not shown, the selected value is still submitted
+    const formSubmitRequest = mockUrqlClient.executeMutation.mock.calls[3][0];
+    const linkedSectionId = formSubmitRequest.variables.widget.section._link;
+    expect(linkedSectionId).toEqual("1");
+  });
+
+  describe("optionsLabel", () => {
+    test("it properly renders options with an `optionsLabel` string parameter", async () => {
+      setupRenderResult({ optionLabel: "customField" });
+      expect(screen.getByText(sectionFooEdge.node.customField)).toBeInTheDocument();
+      expect(screen.getByText(sectionBarEdge.node.customField)).toBeInTheDocument();
+    });
+
+    test("it properly renders options with an `optionsLabel` function parameter", async () => {
+      setupRenderResult({ optionLabel: (record) => `Name: ${record.name} id: ${record.id}` });
+      expect(screen.getByText("Name: Section Foo id: 1")).toBeInTheDocument();
+      expect(screen.getByText("Name: Section Bar id: 2")).toBeInTheDocument();
+    });
+  });
+});
+
+const mockUpdateWidgetFindBy = () => {
+  mockWidgetFindBy(
+    {
+      name: "Update",
+      apiIdentifier: "update",
+      operatesWithRecordIdentity: true,
+    },
+    { id: "42" }
+  );
+  mockSectionQueryResponse();
+};
+
+const mockSectionQueryResponse = (edges?: any[]) => {
+  mockUrqlClient.executeQuery.pushResponse("sections", getSectionsQueryResponse(edges));
+};
+
+const metadata: ActionMetadata = {
+  name: "Widget",
+  action: {
+    name: "Create",
+    apiIdentifier: "create",
+    inputFields: [
```

Comment:
Is it possible to use the action metadata from `packages/react/spec/auto/support/widgetModel.ts` ?

Reply:
Good call. Updated to use the field metadata result from `widgetModel.ts`